### PR TITLE
Support ktx2 supercompressed textures

### PR DESCRIPTION
**Pull request by slimbuck**
This PR:
- adds support for KTX2 basis compressed textures
- a followup PR will add support for non-supercompressed KTX2 formats
- updates the engine basis modules to include ktx2 support (code is taken from basisu version 1.15)
- removes the unused `dds` parser and renames `dds-legacy` to `dds`

### CODE REVIEW

**Comment by willeastcott** [comment]
The Basis runtime seems to be updated here. Want to add details of that to the PR description?

**Comment by slimbuck** [comment]
I mention:

> * updates the engine basis modules to ones with ktx2 load support included

What other details would be useful?

**Comment by willeastcott** [comment]
I mean version info. Did you update to 1.15?

File: src/resources/basis-worker.js [hunk 1]
```diff
@@ -157,8 +157,98 @@ function BasisWorker() {
...
+
+        const compressedFormat = isCompressedFormat(basisFormat);
+
+        return {
+            format: basisToEngineMapping(basisFormat, options.deviceDetails),
+            width: compressedFormat ? ((width + 3) & ~3) : width,
+            height: compressedFormat ? ((height + 3) & ~3) : height,
+            levels: levelData,
+            cubemap: false,
+            mipmaps: true,
```
Comment (willeastcott) [inline src/resources/basis-worker.js:243]: Bit confused about this. So KTX2 won't contain non-mipmapped texture data?

File: src/resources/parser/texture/ktx2.js [hunk 2]
```diff
@@ -0,0 +1,202 @@
+import { http } from '../../../net/http.js';
+import { basisTranscode } from '../../basis.js';
+import { Texture } from '../../../graphics/texture.js';
+import { ADDRESS_CLAMP_TO_EDGE, ADDRESS_REPEAT, TEXHINT_ASSET } from '../../../graphics/constants.js';
+
+// helper class for organized reading of memory
+class ReadStream {
```
Comment (willeastcott) [inline src/resources/parser/texture/ktx2.js:7]: Should this go in the core folder? Any scope for utilizing in any other modules?

**Review by willeastcott — APPROVED** [review]

**Comment by willeastcott** [comment]
Little question about the two DDS parsers. Are we supremely confident that the parser that's been removed is genuinely not used anywhere?

File: src/resources/basis-worker.js (hunk 1 above)
Comment (slimbuck) [reply src/resources/basis-worker.js]: Ah yes nice catch. This mipmap flag is actually ignored (https://github.com/slimbuck/engine/blob/ktx2-support/src/resources/parser/texture/ktx2.js#L97) and shouldn't been here. I've removed it.

File: src/resources/parser/texture/ktx2.js (hunk 2 above)
Comment (slimbuck) [reply src/resources/parser/texture/ktx2.js:7]: Yes I made a similar class for hdr loading and would like to extract this and have both use the same interface. Just didn't want to overcomplicate this PR.

**Comment by slimbuck** [comment]
> I mean version info. Did you update to 1.15?

Oh I see. The version 1.15 actually refers to the version of the basisu tool. I don't believe it applies to the transcoder. It would be great to have versioning there though. I might request that in an issue on the repo.

File: src/resources/parser/texture/dds.js [hunk 3]
```diff
@@ -5,88 +5,17 @@ import {
...
-fccToFormat[FCC_DXT1] = PIXELFORMAT_DXT1;
-fccToFormat[FCC_DXT5] = PIXELFORMAT_DXT5;
-fccToFormat[FCC_ETC1] = PIXELFORMAT_ETC1;
-fccToFormat[FCC_PVRTC_2BPP_RGB_1] = PIXELFORMAT_PVRTC_2BPP_RGB_1;
-fccToFormat[FCC_PVRTC_2BPP_RGBA_1] = PIXELFORMAT_PVRTC_2BPP_RGBA_1;
-fccToFormat[FCC_PVRTC_4BPP_RGB_1] = PIXELFORMAT_PVRTC_4BPP_RGB_1;
-fccToFormat[FCC_PVRTC_4BPP_RGBA_1] = PIXELFORMAT_PVRTC_4BPP_RGBA_1;
-
 /**
- * @private
```
Comment (willeastcott) [inline src/resources/parser/texture/dds.js:85]: Just wondering why the private tag is being removed here. `DdsParser` isn't exported as public API. Does this class now appear in the API referecne docs?

**Pull request merged**